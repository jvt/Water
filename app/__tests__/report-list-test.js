import {
	Text,
} from 'react-native';
import React from 'react';
import ReportList from '../report/list'
import Button from 'apsl-react-native-button';
import Renderer from 'react-test-renderer';

describe('ReportList', () => {
	test('BackButton', () => {
		const instance  = Renderer.create(
			<Button 
			style={{}}>
			Back
			</Button>
		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('NewButton', () => {
		const instance  = Renderer.create(
			<Button 
			style={{}}>
			New
			</Button>
		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('ElementLocationText', () => {
		const instance  = Renderer.create(
			<Text
			style={{}}>
			Location
			</Text>
		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('ElementTypeText', () => {
		const instance  = Renderer.create(
			<Text
			style={{}}>
			Type
			</Text>
		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('ElementConditionText', () => {
		const instance  = Renderer.create(
			<Text
			style={{}}>
			Condition
			</Text>
		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('ElementReportedText', () => {
		const instance  = Renderer.create(
			<Text
			style={{}}>
			Reported
			</Text>
		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

	test('ElementSubmittedByText', () => {
		const instance  = Renderer.create(
			<Text
			style={{}}>
			SubmittedBy
			</Text>
		);

		expect(instance.toJSON()).toMatchSnapshot();
	});

})