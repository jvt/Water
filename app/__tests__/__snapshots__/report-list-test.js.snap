exports[`ReportList BackButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "fontSize": 18,
          "textAlign": "center",
        },
        undefined,
      ]
    }>
    Back
  </Text>
</View>
`;

exports[`ReportList ElementConditionText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Condition
</Text>
`;

exports[`ReportList ElementLocationText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Location
</Text>
`;

exports[`ReportList ElementReportedText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Reported
</Text>
`;

exports[`ReportList ElementSubmittedByText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  SubmittedBy
</Text>
`;

exports[`ReportList ElementTypeText 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={Object {}}>
  Type
</Text>
`;

exports[`ReportList NewButton 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 44,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
          "fontSize": 18,
          "textAlign": "center",
        },
        undefined,
      ]
    }>
    New
  </Text>
</View>
`;
